clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -167,18 +163,38 @@ static ConstAtom *analyzeAllocExpr(CallExpr *CE, Constraints &CS, QualType &ArgT
   return nullptr;
 }
 
-ConstraintVariable *
-ConstraintResolver::getTemporaryConstraintVariable(clang::Expr *E,
-                                                   ConstraintVariable *CV) {
-  auto ExpKey = std::make_pair(E, CV);
-  if (ExprTmpConstraints.find(ExpKey) == ExprTmpConstraints.end()) {
-    // Make a copy and store the copy of the underlying constraint
-    // into TempConstraintVars to handle memory management.
-    auto *CVarPtr = CV->getCopy(Info.getConstraints());
-    TempConstraintVars.insert(CVarPtr);
-    ExprTmpConstraints[ExpKey] = CVarPtr;
+std::set<ConstraintVariable *>
+    ConstraintResolver::getInvalidCastPVCons(Expr *E) {
+  std::set<ConstraintVariable *> Ret;
+  if (hasPersistentConstraints(E)) {
+    return getPersistentConstraints(E,Ret);
+  }
```

Comment:
I can imagine that we might put a check like this in a lot of places. That is, in code in which we generate a fresh constraint for a RHS, we first check if there's a persistent constraint and if so, return that. Is there a way to consolidate such checks  away from the function that generates the constraint, but also without cluttering up the callers? 

For example, could we make `PersistentConstraints` into its own class with a method like this (I'm faking it for the function pointer syntax)?
```
std::set<ConstraintVariarable *> generatePersistentConsVar(Expr *E, fnptr F(E,...)) {
  std::set<ConstraintVariable *> Ret;
  if (hasPersistentConstraints(E)) {
    return getPersistentConstraints(E,Ret);
  }
  getPersistentConstraints(E, F(E));
}
```
Then we add to this class as many methods as we like, which call the above one. E.g., we add `getInvalidCastPVCons(Expr *E)` that makes a function pointer out of `ConstraintResolver::getInvalidCastPVCons(Expr *E)` and then invokes `generatePersistentConsVar` with `E` and that function pointer?